### DISCUSSION TITLE
Issues while using GlueJobOperator

### DISCUSSION THREAD

**Discussion by kumbarsg** [apache/airflow#25998] [General]
1. I am trying to create a glue job with this configuration 'NumberOfWorkers': 2, 'WorkerType': 'G.1X'. Here's my code for job creation.

```
job_name = "glue_job"
    submit_glue_job = AwsGlueJobOperator(
        task_id="glue_job",
        job_name=job_name,
        wait_for_completion=True,
        # num_of_dpus=10,
        retry_limit=0,
        script_location=f"s3://bucket/etl.py",
        s3_bucket=GLUE_EXAMPLE_S3_BUCKET,
        iam_role_name=GLUE_CRAWLER_ROLE.split("/")[-1],
        create_job_kwargs={
            'GlueVersion': '3.0', 'NumberOfWorkers': 2, 'WorkerType': 'G.1X',
            "DefaultArguments": {"--enable-glue-datacatalog": ''}
        }
```
and here's the error:

```
when calling the CreateJob operation: 
Please do not set Allocated Capacity if using Worker Type and Number of Workers
```
I checked the official documentation to see if the Allocated capacity is assigned to any default value, but it's not. here's the source code link for the operator.

----------------------------------------------------------------------------------------------------------------------------------

2.  Also, I wonder what is going wrong when I use import GlueJobOperator, instead I have to use AwsGlueJobOperator. My MWAA says it is not found. I think it is bcoz of airflow version 2.2.2. Am I right? Help me understand this perfectly.

**Comment by asif3051** [comment]
HI @kumbarsg did you find the fix the issue?I am trying to import AWS glue job operator but it always give me the import error. And when use GlueJobOperator, then the s3 script doesn't work properly thworing error about some ascii character issue.
How can  I import AWSGlueJobOperator?i am using lter 7.x version of operators

**Reply by kumbarsg** [reply]
- Hi @asif3051  , I didnt find any fix to that issue but did some work around those issues like using older operator name 'AwsGlueJobOperator' to import. Coming to s3 script, it might be because of some unwanted charecters in your code maybe, the ascii charecter issue shouldnt be with operators. Also, few other pointers to these operators are below:
These operators do not update the properties of the crawler and jobs while running. This was confirmed by Airflow. Here is the link to the issue: (https://github.com/apache/airflow/issues/25996) 

These operators work well in a local environment of airflow for create operation but the update operation is still an issue with them.
